pkg/collectors/pool.go

```diff
@@ -310,41 +440,94 @@ func createLogicalCapacityPoolMetrics(ch chan<- prometheus.Metric, f *PerfCollec
+	//logicalUsedCapacity := totalLogicalCapacity - logicalUsableCapacity - childPoolCapacity
 
-	logicalUsedCapacity := logicalCapacity - logicalUsableCapacity
+	newPoolCapacityMetrics(ch, f.poolDescriptors[PoolLogicalCapacityUsable], logicalUsableCapacity, &poolInfo)
 	newPoolCapacityMetrics(ch, f.poolDescriptors[PoolLogicalCapacityUsed], logicalUsedCapacity, &poolInfo)
+	newPoolCapacityMetrics(ch, f.poolDescriptors[PoolLogicalCapacity], totalLogicalCapacity, &poolInfo)
 }
 
-func createPhysicalCapacityPoolMetrics(ch chan<- prometheus.Metric, f *PerfCollector, pool Pool, poolInfo PoolInfo) {
+func createPhysicalCapacityPoolMetrics(ch chan<- prometheus.Metric, f *PerfCollector, pool Pool, poolInfo PoolInfo, fsRestClient *rest.FSRestClient, mDisksList rest.MDisksList) {
 	if isParentPool(pool) {
+		var reclaimableCalculatedCapacity float64
 		physicalFree, err := strconv.ParseFloat(pool[PhysicalFreeKey].(string), 64)
 		if err != nil {
 			log.Errorf("get physical free failed: %s", err)
 			return
 		}
-
-		reclaimable, err := strconv.ParseFloat(pool[ReclaimableKey].(string), 64)
+		physical, err := strconv.ParseFloat(pool[PhysicalCapacityKey].(string), 64)
 		if err != nil {
-			log.Errorf("get reclaimable failed: %s", err)
+			log.Errorf("get physical capacity failed: %s", err)
 			return
 		}
-
-		physical, err := strconv.ParseFloat(pool[PhysicalCapacityKey].(string), 64)
+		poolOrigReclaimable, err := strconv.ParseFloat(pool[ReclaimableKey].(string), 64)
 		if err != nil {
-			log.Errorf("get physical capacity failed: %s", err)
+			log.Errorf("get reclaimable failed: %s", err)
 			return
-
 		}
-		usable := physicalFree + reclaimable
-		used := physical - usable
+		if poolOrigReclaimable != 0 {
+			reclaimableCalculatedCapacity, err = f.GetPoolReclaimablePhysicalCapacity(pool, fsRestClient, mDisksList)
+			if err != nil {
+				log.Errorf("get reduced reclaimable capacity failed: %s", err)
+				return
+			}
+		} else {
+			reclaimableCalculatedCapacity = 0
+		}
+
+		used := physical - physicalFree - reclaimableCalculatedCapacity
+		usable := physicalFree + poolOrigReclaimable
+
 		newPoolCapacityMetrics(ch, f.poolDescriptors[PoolCapacityUsable], usable, &poolInfo)
 		newPoolCapacityMetrics(ch, f.poolDescriptors[PoolCapacityUsed], used, &poolInfo)
+		newPoolCapacityMetrics(ch, f.poolDescriptors[PoolPhysicalCapacity], physical, &poolInfo)
+	} else {
+		newPoolCapacityMetrics(ch, f.poolDescriptors[PoolCapacityUsable], InvalidVal, &poolInfo)
+		newPoolCapacityMetrics(ch, f.poolDescriptors[PoolCapacityUsed], InvalidVal, &poolInfo)
+		newPoolCapacityMetrics(ch, f.poolDescriptors[PoolPhysicalCapacity], InvalidVal, &poolInfo)
+	}
+}
+
+func (f *PerfCollector) GetPoolReclaimablePhysicalCapacity(pool Pool, fsRestClient *rest.FSRestClient, mDisksList rest.MDisksList) (float64, error) {
+	var reclaimable float64
+
+	var dataReduction bool
+	if pool[DataReductionKey].(string) == "yes" {
+		dataReduction = true
+	}
```

Comment:
I think we can reduce to one line (and still keep it readable)
`dataReduction := pool[DataReductionKey].(string) == "yes"`

Reply:
fixed.